### PR TITLE
#165273587 Social media article sharing

### PR DESCRIPTION
**Pull request by llwasampijja**
#### What does this PR do?
Allows an authenticated user to share an article on social media
#### Description of Task to be completed?
An authenticated user can share an article on facebook, twitter or email
#### How should this be manually tested?
It can be tested in postman:
* email share endpoint: `http://127.0.0.1:8000/api/articles/article_id/emailshare/`
* facebook share endpoint: `http://127.0.0.1:8000/api/articles/article_id/facebookshare/`
* twitter share endpoint: `http://127.0.0.1:8000/api/articles/article_id/twittershare/`
#### What are the relevant pivotal tracker stories?
[#165273587](https://www.pivotaltracker.com/story/show/165273587)
#### Screenshots (if appropriate)

### CODE REVIEW

File: authors/apps/articles/tests/test_aticle_share.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        self.client.force_authenticate(user=self.user)
+
+        response = self.client.post(self.articles_endpoint, data=article_data, format='json')
+        self.article_id_1 = response.data.get("article").get("id")
+        self.facebook_share_url = f"{self.articles_endpoint}{self.article_id_1}/facebookshare/"
+        self.twitter_share_url = f"{self.articles_endpoint}{self.article_id_1}/twittershare/"
+        self.email_share_url = f"{self.articles_endpoint}{self.article_id_1}/emailshare/"
+
+    def test_share_on_facebook(self):
+         self.assertEqual(201, status.HTTP_201_CREATED)
```
Comment (bisonlou) [inline authors/apps/articles/tests/test_aticle_share.py]: ```self.assertEqual(201, status.HTTP_201_CREATED)```
Is this really necessary?

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -238,9 +244,82 @@ def delete(self, request, **kwargs):
...
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (JSONRenderer, )
+
+    def post(self, request, *args, **kwargs):
+        server_url = f"https://{get_current_site(request)}"
+        articles_endpoint = 'api/articles'
+        twitter_share_link = f"{server_url}/{articles_endpoint}/{kwargs.get('article_id')}/"
+        twitter_share_url = "https://twitter.com/home?status="
+        message = {"twitter_share_link": f"{twitter_share_url}{twitter_share_link}"}
+        return Response(message)
```
Comment (bisonlou) [inline authors/apps/articles/views.py]: I notice some duplication in the two post methods under ShareOnFacebookView and ShareOnTwitterView. Could those two methods be merged into one?

File: authors/apps/articles/views.py [hunk 3]
```diff
@@ -238,9 +244,82 @@ def delete(self, request, **kwargs):
...
+    def create(self, request, *args, **kwargs):
+        article = Article.objects.get(id=self.kwargs.get("article_id"))
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        reciever_mail = serializer.data['email']
+        user_obj = User.objects.get(username=request.user.username)
+        user_serializer = UserSerializer(user_obj)
+        article_serializer = ArticleSerializer(article)
+        article_id = article.id
+        sender_email = request.user.email
```
Comment (bisonlou) [inline authors/apps/articles/views.py:317]: This block of code is quite hard to read. Please group similar function statements with a line break.

**Review by bisonlou — CHANGES_REQUESTED** [review]

**Review by bisonlou — APPROVED** [review]

**Review by nadralia — APPROVED** [review]
Nice Work @llwasampijja

**Review by kalsmic — APPROVED** [review]
well done, @llwasampijja.

**Pull request merged**